src/components/PathwayCourse/C4CApathway/index.js

```diff
@@ -0,0 +1,219 @@
+import React, { useEffect, useState } from "react";
+import {
+  Container,
+  Grid,
+  Typography,
+  Box,
+  Card,
+  CardContent,
+  CardActions,
+} from "@mui/material";
+import { METHODS } from "../../../services/api";
+import axios from "axios";
+import { useSelector, useDispatch } from "react-redux";
+import useStyles from "../styles";
+import { PATHS, interpolatePath } from "../../../constant";
+import { Link, useHistory } from "react-router-dom";
+import { LinearProgress } from "@mui/material";
+import LockIcon from "@mui/icons-material/Lock";
+import LockOpenIcon from "@mui/icons-material/LockOpen";
+import { useMediaQuery } from "@mui/material";
+import { breakpoints } from "../../../theme/constant";
+
+function C4CApathway() {
+  const user = useSelector(({ User }) => User);
+  const [pathway, setPathway] = useState([]);
+  const [completedPortion, setCompletedPortion] = useState({});
+  const classes = useStyles();
+  let completedAll = completedPortion?.total === 100;
+  const isActive = useMediaQuery("(max-width:" + breakpoints.values.sm + "px)");
+
+  useEffect(() => {
+    axios({
+      method: METHODS.GET,
+      url: `${process.env.REACT_APP_MERAKI_URL}/pathways/c4ca`,
+      headers: {
+        accept: "application/json",
+        Authorization: user?.data?.token,
+      },
+    })
+      .then((response) => {
+        setPathway(response?.data);
+      })
+      .catch((err) => {});
+  }, [setPathway]);
+
+  useEffect(() => {
+    axios({
+      method: METHODS.GET,
+      url: `${process.env.REACT_APP_MERAKI_URL}/pathways/${pathway.id}/completePortion`,
+      headers: {
+        accept: "application/json",
+        Authorization: user?.data?.token,
+      },
+    })
+      .then((response) => {
+        setCompletedPortion((prevState) => ({
+          ...prevState,
+          total: response?.data?.total_completed_portion,
+        }));
+
+        response.data.pathway.map((item) => {
+          setCompletedPortion((prevState) => ({
+            ...prevState,
+            [item.course_id]: item.completed_portion,
+          }));
+        });
+      })
+      .catch((err) => {});
+  }, [pathway]);
+  // console.log(completedPortion)
+
+  const filterCourses = pathway?.modules?.filter((item) => {
+    return item.courses.length > 0;
+  });
+
+  const [unlockedCourses, setUnlockedCourses] = useState([]);
+
+  // Function to check if a course is unlocked
+
+  // Effect to automatically unlock courses when their percentage reaches 100
+  useEffect(() => {
+    filterCourses &&
+      filterCourses.forEach((item, index) => {
+        const isLastCourse = item.courses.length - 1;
```

Comment:
Yeh variable ka naam `lastCourseIndex` ko baadalen kyonki yeh Boolean nahi hai.
```suggestion
        const lastCourseIndex = item.courses.length - 1;
```